iterating-over-data/practice-iterating-over-data.md

```diff
@@ -0,0 +1,465 @@
+* `for i in range(1,23)`
+
+##### !end-options
+
+##### !answer
+
+`for i in range(1,23)`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+Python's `range` has an *exclusive nature*, meaning that it does not include the last integer in its range, ie the `stop` parameter. That's why we must use `23` instead of just `22`.
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: EkdvU6Z0K
+* title: Iterate Over a List
+
+##### !question
+
+Create a function named `sum_list` that takes a parameter `list` (a list of integers). Iterate through the list and sum up each integer. Then return the total.
+
+##### !end-question
+
+##### !placeholder
+
+```python
+
+def sum_list(list):
+  pass
+
+```
+
+##### !end-placeholder
+
+##### !tests
+```python
+
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+def test_sum_list(self):
+  self.assertEqual(sum_list([3,5,7,9]),24)
+  self.assertEqual(sum_list([0,-3,6,6]),9)
+
+```
+##### !end-tests
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+An example of a working implementation:
+
+``` python
+
+def sum_list(list):
+  sum = 0
+  for i in list:
+    sum += i
+  return sum
+
+```
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: checkbox
+* id: N1iCy1fCF
+* title: Choose the Best Option(s)
+
+
+##### !question
+
+Given this dictionary: 
+``` python
+
+my_profile = {"icecream_flavor": "chocolate", has_siblings: True}
+
+```
+
+Which loop(s) *can* be used to print out each key?
+
+##### !end-question
+
+##### !options
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+
+##### !end-options
+
+##### !answer
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x, y in my_profile.items()`
+
+##### !end-answer
+
+
+##### !hint
+
+##### !end-hint
+
+
+##### !rubric
+
+##### !end-rubric
+
+
+##### !explanation
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 4 -->
+<!-- prettier-ignore-start -->
+### !challenge
+
+* type: checkbox
+* id: NyhIeJG0K
+* title: Choose the Best Option(s)
+
+
+##### !question
+
+Given this dictionary: 
+``` python
+
+my_profile = {"icecream_flavor": "chocolate", has_siblings: True}
+
+```
+
+Which loop(s) *can* be used to print out each value?
+
+##### !end-question
+
+##### !options
+
+* `for x in my_profile:`
+* `for x in my_profile.keys()`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
+
+
+##### !end-options
+
+##### !answer
+
+* `for x in my_profile:`
+* `for x in my_profile.values()`
+* `for x, y in my_profile.items()`
```

Comment:
This answer should have all of them, technically? I have a note similar to this below